File: app/dtos/chatbot.py
from enum import StrEnum

from pydantic import BaseModel


class ChatbotContextType(StrEnum):
    MAIN = "MAIN"
    ANALYSIS = "ANALYSIS"
    DIET = "DIET"
    CHALLENGE = "CHALLENGE"
    GENERAL = "GENERAL"


class ChatbotAskRequest(BaseModel):
    message: str
    context_type: ChatbotContextType = ChatbotContextType.GENERAL
    target_id: int | None = None


class ChatbotAskResponse(BaseModel):
    answer: str
    source: str = "DUMMY_LLM"
    context_type: ChatbotContextType
    recommended_actions: list[str]
    safety_notice: str

File: frontend/src/api/chatbot.ts
import { apiRequest } from "./client";

export type ChatbotContextType = "MAIN" | "ANALYSIS" | "DIET" | "CHALLENGE" | "GENERAL";

export type ChatbotAskRequest = {
  message: string;
  context_type?: ChatbotContextType;
  target_id?: number | null;
};

export type ChatbotAskResponse = {
  answer: string;
  source: "DUMMY_LLM" | string;
  context_type: ChatbotContextType;
  recommended_actions: string[];
  safety_notice: string;
};

export async function askChatbot(payload: ChatbotAskRequest): Promise<ChatbotAskResponse> {
  return apiRequest<ChatbotAskResponse>("/chatbot/ask", {
    method: "POST",
    body: payload,
  });
}
